modules/widgets/wordpress-post-widget.php

```diff
@@ -926,7 +926,12 @@ public function form( $instance ) {
 				<?php _e( "Enter a WordPress.com or Jetpack WordPress site URL.", 'jetpack' ); ?>
 			</i>
 			<?php
-			if ( empty( $url ) ) {
+			/**
+			 * Show an error if the URL field was left empty.
+			 *
+			 * The error is shown only when the widget was already saved.
+			 */
+			if ( empty( $url ) && ! preg_match( '/__i__|%i%/', $this->id ) ) {
```

Comment:
You can check `$this->updated`. I tested replacing the regular expression with
```php
if ( empty( $url ) && $this->updated ) {
```
and it works fine.
From documentation in `WP_Widget` class for `$this->updated`:
> Whether the widget data has been updated.
> Set to true when the data is updated after a POST submit - ensures it does not happen twice.